base/REPLCompletions.jl

```diff
@@ -161,7 +161,13 @@ function complete_path(path::AbstractString, pos; use_envpath=false)
                 continue
             end
 
-            local filesinpath = readdir(pathdir)
+            local filesinpath
+            try
+                filesinpath = readdir(pathdir)
+            catch
+                # Bash allows dirs in PATH that can't be read, so we should as well.
```

Comment:
the only way to check if something is readable is to read it (http://linux.die.net/man/2/access)